pkg/cli/alpha/generate.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+	http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package alpha
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/kubebuilder/v3/pkg/config/store"
+	"sigs.k8s.io/kubebuilder/v3/pkg/config/store/yaml"
+	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugin/util"
+)
+
+const defaultOutputDir = "output-dir"
+
+// NewScaffoldCommand return a new scaffold command
+func NewScaffoldCommand() *cobra.Command {
+	var outputPath, inputPath string
+	scaffoldCmd := &cobra.Command{
+		Use:   "generate",
+		Short: "Re-scaffold an existing kuberbuilder project",
+		Long:  `Re-scaffold an existing kuberbuilder project`,
+		Run: func(cmd *cobra.Command, args []string) {
+			cwd, err := os.Getwd()
+			if err != nil {
+				log.Fatal(err)
+			}
+			inputPath := getInputPath(cwd, inputPath)
+			projectPath := fmt.Sprintf("%s/%s", inputPath, yaml.DefaultPath)
+			store := yaml.New(machinery.Filesystem{FS: afero.NewOsFs()})
+			if err = store.LoadFrom(projectPath); err != nil {
+				log.Fatal(err)
+			}
+			outputDirectory := getOutputPath(cwd, outputPath)
+			// create output directory
+			if err = os.MkdirAll(outputDirectory, os.ModePerm); err != nil {
+				log.Fatal(err)
+			}
+			// use the new directory to set up the new project
+			if err = os.Chdir(outputDirectory); err != nil {
+				log.Fatal(err)
+			}
+			// change back to the cwd after completion
+			defer func() {
+				_ = os.Chdir(cwd)
+			}()
+			// init project with plugins
+			if err = kubebuilderInit(store); err != nil {
+				log.Fatal(err)
+			}
+		},
+	}
+	scaffoldCmd.Flags().StringVar(&inputPath, "input-dir", "",
+		"path to a kubebuilder project file if not in the current working directory")
+	scaffoldCmd.Flags().StringVar(&outputPath, "output-dir", defaultOutputDir,
+		"path to output the scaffolding. defaults to the current working directory")
+	return scaffoldCmd
+}
+
```

Comment:
I think we need a func like setDefaults which will set the defaults ([here](https://github.com/operator-framework/operator-sdk/blob/5347d9375658cab9117e7ac8f691d35bb9154ff9/internal/cmd/operator-sdk/generate/bundle/bundle.go#L116-L121) you can find a nice example)

Also, one validate() which we should call before do anything and has the purpose to validate if ALL is OK prior run the command or fail. What do you think?

Reply:
Cool! I used to have a validate() in it, but I don't know what it should check so I delete it. Maybe it can check 'whether it can find the kubebuilder binary'. :)

Reply:
I think you must ensure that has a PROJECT file
Otherwise, how will you do the re-scaffold? 

For now, I think just that. But we will need also do checks like
If the Project file has layout == go/v2 or go/v3 AND if the Project file is the old config version than we say that the command is not supported. but we can do that in a follow up.

Reply:
Thanks! So far my code will ensure the PROJECT file.
I will add other check in a follow up. :)